app/layouts/DashboardLayout/OnboardingModal.view/OnboardingModal.test.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import '../../../testUtils/i18nForTests';
+
+import { OnboardingModal } from './OnboardingModal.view';
+
+const confirmEntropyKnown = jest.fn();
+const setPin = jest.fn();
+const pendingSecrets = null;
+const showEntropyMsg =
+  'We generated a random Entropy to create your new account. Please store this code in a secure, private manner. You will need your Entropy to import this account into other wallets.';
+const setPinMsg =
+  'Your Transaction PIN is used as an extra safety measure. It is required if a transaction value is higher than the threshold MOB amount.';
+
+describe('OnboardingModal', () => {
+  test('renders ShowEntropyModal when no stored entropy', () => {
+    const { getByText } = render(
+      <OnboardingModal
+        confirmEntropyKnown={confirmEntropyKnown}
+        setPin={setPin}
+        pendingSecrets={pendingSecrets}
+        isEntropyKnown={false}
+        isPinRequired
+      />
+    );
+
+    expect(getByText(showEntropyMsg)).toBeInTheDocument();
+  });
+
+  test('renders SetPinModal when pin is required', () => {
+    const { getByText } = render(
```

Comment:
The render logic --which is repeated several times-- could be refactored to a function.

Reply:
It's called 3 times, with a varying property on each call - I think it's ok.

Reply:
Pass the different value as a parameter to the function, and that will solve it.